### PR TITLE
Add workflows

### PR DESCRIPTION
**Pull request by rgildein**
- add GitHub workflow to run lint and unit tests

based on: #2

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    branches: [ main ]
+
+jobs:
+  lint-unittests:
+    name: Lint and Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [3.6, 3.8, "3.10"]
```
Comment (Pjack) [inline .github/workflows/tests.yaml]: will 3.10 be interpreted to 3.1 (float) without quotation marks? , I prefer to align the format between  3.6/3.8/3.10, that is "all of them have quotaion mark". But it is not a big problem.

Comment (rgildein) [reply .github/workflows/tests.yaml]: Yes, it will be interpreted as 3.1. I'll add all of them into quotation marks.

**Review by Pjack — CHANGES_REQUESTED** [review]
- I prefer not to use "tests" as the filename. Some people may delete it due to misunderstanding. we can align charmed-kubernetes' naming. unit-lint.yaml
https://github.com/charmed-kubernetes/workflows/tree/main/.github/workflows

- It's a good practice to reuse workflow in each project. We can do it in the future.
e.g:  https://github.com/charmed-kubernetes/charm-kubernetes-worker/blob/main/.github/workflows/main.yml

**Comment by rgildein** [comment]
> * I prefer not to use "tests" as the filename. Some people may delete it due to misunderstanding. we can align charmed-kubernetes' naming. unit-lint.yaml
>       https://github.com/charmed-kubernetes/workflows/tree/main/.github/workflows

I renamed it, you are right it make more sense with proper name.
> 
>     * It's a good practice to reuse workflow in each project. We can do it in the future.
>       e.g:  https://github.com/charmed-kubernetes/charm-kubernetes-worker/blob/main/.github/workflows/main.yml

I think that this is really good idea, but it's definitely out of scope of this PR.

**Review by Pjack — APPROVED** [review]

**Review by agileshaw — APPROVED** [review]
LGTM

**Pull request merged**